plugins/schedule/bulk_upload/batch/BulkUploadScheduleEventCsvEngine.php

```diff
@@ -0,0 +1,435 @@
+	 * @param $eventType
+	 */
+	protected function initTemplateEntry ($templateEntryType)
+	{
+		switch ($templateEntryType)
+		{
+			case self::TEMPLATE_ENTRY_TYPE_MEDIA:
+				$templateEntry = new KalturaMediaEntry();
+				$templateEntry->mediaType = KalturaMediaType::VIDEO;
+				break;
+			case self::TEMPLATE_ENTRY_TYPE_LIVE:
+				$templateEntry = new KalturaLiveStreamEntry();
+				$templateEntry->mediaType = KalturaMediaType::LIVE_STREAM_FLASH;
+				$templateEntry->sourceType = KalturaSourceType::LIVE_STREAM;
+				break;
+		}
+		
+		return $templateEntry;
+	}
+	
+	/**
+	 * Parses out the recurrence string of the following format:
+	 * FREQ=,INTERVAL=,BYMONTHDAY=, BYMONTH=, BYDAY=, BYSETPOS=, COUNT=
+	 * @param string $recurrence
+	 * @param string $endTime
+	 *
+	 * @return KalturaScheduleEventRecurrence
+	 */
+	protected function createRecurrenceObject ($recurrence, $endTime = null)
+	{
+		$recurrencePattern = explode(',', $recurrence);
+		
+		$recurrenceObject = new KalturaScheduleEventRecurrence();
+		foreach ($recurrencePattern as $part)
+		{
+			list ($key, $value) = explode('=', trim($part));
+			
+			switch ($key)
+			{
+				case 'FREQ':
+					$recurrenceObject->frequency = constant("KalturaScheduleEventRecurrenceFrequency::$value");
+					break;
+				case 'INTERVAL':
+					$recurrenceObject->interval = $value;
+					break;
+				case 'BYMONTHDAY':
+					$recurrenceObject->byMonthDay = $value;
+					break;
+				case 'BYMONTH':
+					$recurrenceObject->byMonth = $value;
+					break;
+				case 'BYDAY':
+					$recurrenceObject->byDay = $value;
+					break;
+				case 'BYSETPOS':
+					$recurrenceObject->byOffset = $value;
+					break;
+				case 'COUNT':
+					$recurrenceObject->count = $value;
+					break;
+			}
+		}
+		
+		if($endTime)
+		{
+			$recurrenceObject->count = null;
+			$recurrenceObject->until = strtotime($endTime);
+		}
+		
+		KalturaLog::info('Recurrence object: ' . print_r($recurrenceObject, true));
+		return $recurrenceObject;
+	}
+	
+	/**
+	 * @param $eventType
+	 * @return KalturaEntryScheduleEvent
+	 */
+	protected function initEventObject($eventType)
+	{
+		switch ($eventType)
+		{
+			case KalturaScheduleEventType::RECORD:
+				return new KalturaRecordScheduleEvent();
+			case KalturaScheduleEventType::LIVE_STREAM:
+				return new KalturaLiveStreamScheduleEvent();
+			default:
+				KalturaLog::err("Invalid scheduled event type: $eventType!");
+				return null;
+		}
+	}
+	
+	/**
+	 *
+	 * Creates a new upload result object from the given parameters
+	 * @param array $values
+	 * @param array $columns
+	 * @return KalturaBulkUploadResult
+	 */
+	protected function createUploadResult($values, $columns)
+	{
+		$result = parent::createUploadResult($values, $columns);
+		if (!$result)
+		{
+			return;
+		}
+		
+		$result->bulkUploadResultObjectType = KalturaBulkUploadObjectType::SCHEDULE_EVENT;
+		
+		/* @var $result KalturaBulkUploadResultScheduleEvent */
+		//Input validation
+		$row = array_combine ($columns, $values);
```

Comment:
Worth adding array trim on the columns & values to avoid cases where someone passes "title " for example and we fail to parse it.
Also before ruuning array combine worth checking they are the same length otherwise you will get a mismatch between column and value.